File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& questions) {
        int n = questions.size();
        vector<long long> dp(n, 0ll);
        dp[n-1] = questions[n-1][0];

        for(int i=n-2; i>=0; i--){
            int lookAfter = i+questions[i][1]+1;
            if(lookAfter <= n-1){
                dp[i] = max((long long)dp[i+1], (long long)questions[i][0]+dp[lookAfter]);
            }
            else{
                dp[i] = max((long long)dp[i+1], (long long)questions[i][0]);
            }
        }
        return dp[0];
    }
};

File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.rs
impl Solution {
    pub fn most_points(questions: Vec<Vec<i32>>) -> i64 {
        let n = questions.len();
        let mut dp = vec![0i64; n];

        dp[n-1] = questions[n-1][0] as i64;

        for i in (0..n-1).rev() {
            let look_after = i + questions[i][1] as usize + 1;
            if(look_after < n){
                dp[i] = dp[i+1].max(questions[i][0] as i64 + dp[look_after]);
            }
            else{
                dp[i] = dp[i+1].max(questions[i][0] as i64);
            }
        }

        dp[0]
    }
}